src/compiler/compiler-core/src/compiler.ts

```diff
@@ -147,9 +149,31 @@ export abstract class Compiler<
       ...property,
       initializer,
       type: name,
+      presentation: this.getPropertyPresentation(property, instance),
     };
   }
 
+  protected getPropertyPresentation (property: Property, instance: any): TargetPropertyPresentation {
+    const presentation: TargetPropertyPresentation = {
+      value: presentProperty(instance),
+      reference: '',
+      properties: presentProperties({...instance.defaults, ...instance.overrides}),
+    };
+
+    for (const reference of property.references) {
+      const parent = reference.parentType;
+
+      if (!reference.path.length) {
+        presentation.reference = `${parent}.${reference.name}`;
+        continue;
+      }
+
+      // TODO: add support for nested references.
```

Comment:
@roperzh just for my understanding — can you share an example of a nested reference (ideal) vs how it's presented today?

Reply:
@zackbrown sure! and I think this comment should be more clear. A good example of a component that can have nested references is a `Typograph`, let's say you have the following:

```ts
const colors = {
  black: Color.hex('#101010'),
}

const palette = {
  text: colors.black,
}

const typographs = {
  heading1: new Typograph({color: palette.text}),
}
```

Right now, in the `heading1` comment we show the color value without the reference:

```ts
/**
 * color: #101010
 * ...
 */
```

Ideally we should show `palette.text` next to the value:

```ts
/**
 * color: `palette.text` (#101010)
 * ...
 */
```

I made the call to punt in favor of delivering most of the value of this feature without adding more complexity, but this could be implemented for sure as part of this PR. What do you think?

Reply:
Agreed that punting is the right call; can definitely come back for this if warranted.  Thanks for the explanation.